src/main/java/mServer/crawler/AddToFilmlist.java

```diff
@@ -5,176 +5,270 @@
+            vonListe.addAll(localList);
+            localList.clear();
+            treffer += t.getTreffer();
+        }
+        return treffer;
+    }
 
-    private class AddOld implements Runnable {
+    private class ImportOldFilmlistThread extends Thread {
 
-        private DatenFilm film;
         private final ListeFilme listeOld;
-        private final int MIN_SIZE_ADD_OLD = 5; //REST eh nur Trailer
+        private final ArrayList<DatenFilm> localAddList = new ArrayList<>((vonListe.size() / NUMBER_OF_THREADS) + 500);
+        private int treffer = 0;
+        private OkHttpClient client = null;
 
-        public AddOld(ListeFilme listeOld) {
+        public ImportOldFilmlistThread(ListeFilme listeOld, OkHttpClient client) {
             this.listeOld = listeOld;
             threadCounter.incrementAndGet();
+            this.client = client;
+        }
+
+        public int getTreffer() {
+            return treffer;
+        }
+
+        public ArrayList<DatenFilm> getLocalAddList() {
+            return localAddList;
+        }
+
+        private void addOld(DatenFilm film) {
+            treffer++;
+            film.init();
+
+            localAddList.add(film);
+        }
+
+        private synchronized DatenFilm popOld(ListeFilme listeOld) {
+            if (!listeOld.isEmpty()) {
+                return listeOld.remove(0);
+            } else
+                return null;
         }
 
         @Override
         public void run() {
-            EtmPoint performancePoint = etmMonitor.createPoint("AddOld:run");
-            while (!stopOld && (film = popOld(listeOld)) != null) {
-                long size = FileSize.laengeLong(film.arr[DatenFilm.FILM_URL]);
-                if (size > MIN_SIZE_ADD_OLD) {
-                    addOld(film);
+
+            DatenFilm film;
+            while (!isInterrupted() && (film = popOld(listeOld)) != null) {
+                final String url = film.arr[DatenFilm.FILM_URL];
+                if (film.arr[DatenFilm.FILM_GROESSE].isEmpty()) {
+                    Request request = new Request.Builder().url(url).head().build();
+                    try (Response response = client.newCall(request).execute();
+                         ResponseBody body = response.body()) {
+                        if (response.isSuccessful()) {
+                            long respLength = body.contentLength();
+                            if (respLength < 1_000_000) {
+                                respLength = -1;
+                            } else if (respLength > 1_000_000) {
+                                respLength /= 1_000_000;
+                            }
+
+                            if (respLength > MIN_SIZE_ADD_OLD)
+                                addOld(film);
+                        }
+                    } catch (SocketTimeoutException ignored) {
+                    } catch (IOException ex) {
+                        ex.printStackTrace();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Use a logger to log this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1148)